### PR TITLE
Layer refactor and interactivity

### LINKED ISSUE #118: Controls after adding mesh still orbit around origin
Annoying that I have to search for the mesh I just put in

### PR DESCRIPTION
**Pull request by Raphtor**
Fixes #118 in favor of #119 

Refactors into register-able layers. Now, users can register their own functions with the figure. This also allows us to have different interactivity callbacks, which has also been added.

Example:
```python
import pytri
import numpy as np


fig = pytri.Figure()
fig.register_layer(layers.ImshowLayer)
im = fig.imshow(np.random.random((100,100)))
im.on_click = lambda picker: f'Beep @ {picker.point}'
fig.show()

```

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "/Users/normar1/.virtualenvs/microns/bin/python"
```
Comment (j6k4m8) [inline .vscode/settings.json:2]:
```suggestion

```

**Review by j6k4m8 — APPROVED** [review]
Cool! I really like the registration pattern where layers can declare their own preferred name.

Is there a way to override this if you want to register them under a different name? For example,

```python
from my_special_library import BigDataImageLayer

fig.register_layer("imshow", BigDataImageLayer)
```

Or do we make the assumption that you'll need to pick a name without collisions?

**Comment by Raphtor** [comment]
Good point. Added an option, default to use class name.

**Review by j6k4m8 — APPROVED** [review]

**Comment by j6k4m8** [comment]
At some point this should go into the Changelog and we should bump the version marker!

**Pull request merged**